### PR TITLE
[IndexBundle] Migrate to SettingsStoreAwareInstaller

### PR DESCRIPTION
**Pull request by aarongerig**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no

This PR implements the `SettingsStoreAwareInstaller` for the `CoreShopIndexBundle`.

### CODE REVIEW

File: src/CoreShop/Bundle/IndexBundle/Installer.php [hunk 1]
```diff
@@ -46,33 +47,21 @@ public function install(): void
...
 
+    /**
+     * {@inheritdoc}
+     */
     public function needsReloadAfterInstall(): bool
     {
         return true;
     }
 
+    /**
```
Comment (dpfaffenbauer) [inline src/CoreShop/Bundle/IndexBundle/Installer.php:62]: remove doc blocks

File: src/CoreShop/Bundle/IndexBundle/Installer.php [hunk 2]
```diff
@@ -15,16 +15,17 @@
...
 use Pimcore\Console\Application;
-use Pimcore\Extension\Bundle\Installer\InstallerInterface;
+use Pimcore\Extension\Bundle\Installer\SettingsStoreAwareInstaller;
 use Symfony\Component\Console\Input\ArrayInput;
 use Symfony\Component\Console\Output\NullOutput;
 use Symfony\Component\Console\Output\OutputInterface;
 use Symfony\Component\HttpKernel\KernelInterface;
 
-class Installer implements InstallerInterface
+class Installer extends SettingsStoreAwareInstaller
```
Comment (dpfaffenbauer) [inline src/CoreShop/Bundle/IndexBundle/Installer.php:24]: what about the migration? bundle should be marked installed with a migration

**Review by dpfaffenbauer — CHANGES_REQUESTED** [review]

File: src/CoreShop/Bundle/IndexBundle/Installer.php (hunk 2 above)
Comment (solverat) [reply src/CoreShop/Bundle/IndexBundle/Installer.php:24]: Why? Just do a `parent::install()` in your `install()` method which will execute the initial setting-store stuff.

Comment (dpfaffenbauer) [reply src/CoreShop/Bundle/IndexBundle/Installer.php:24]: I am talking about already installed IndexBundles. With this change, these would not be marked as installed anymore

Comment (dpfaffenbauer) [reply src/CoreShop/Bundle/IndexBundle/Installer.php:24]: True... I think we can then ignore it

Comment (aarongerig) [reply src/CoreShop/Bundle/IndexBundle/Installer.php:24]: Sorry, just to make it clear: By "ignore it" you mean, remove the migration? 😅

Comment (dpfaffenbauer) [reply src/CoreShop/Bundle/IndexBundle/Installer.php:24]: @aarongerig Since you added it, we can keep it now :) thanks for that

**Review by dpfaffenbauer — APPROVED** [review]

**Comment by dpfaffenbauer** [comment]
thanks a lot @aarongerig

**Pull request merged**